src/radical/pilot/agent/launch_method/ibrun.py

```diff
@@ -41,38 +39,52 @@ def construct_command(self, cu, launch_script_hop):
 
         slots        = cu['slots']
         cud          = cu['description']
+
         task_exec    = cud['executable']
-        task_cores   = cud['cpu_processes']  # FIXME: handle cpu_threads
         task_args    = cud.get('arguments') or []
         task_argstr  = self._create_arg_string(task_args)
+        task_env     = cud.get('environment') or dict()
+
+        n_tasks      = cud['cpu_processes']  # FIXME: handle cpu_threads
 
-        cpn     = self._cfg.cores_per_node
-        index   = 0
-        offsets = list()
+        # Usage of env variable TACC_TASKS_PER_NODE is purely for MPI tasks,
+        #  and threads are not considered (info provided by TACC support)
+        n_node_tasks = int(task_env.get('TACC_TASKS_PER_NODE') or
+                           self._cfg.get('cores_per_node', 1))
+
+        if not self._node_list:
+            raise RuntimeError('attribute "_node_list" (from RM) is not set')
```

Comment:
We check the validity of the node list during the [RM initialization](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/agent/resource_manager/base.py#L116), and thus this line is really just checking if the constructor gets that node list from the configuration.  Should we move that test into the constructor, if you think we should have it?

Reply:
yes, totally agree, thank you! It is redundant to have this validity check here (will update this as well)